### DISCUSSION TITLE
403 Forbidden nginx/1.18.0 (Ubuntu)

### DISCUSSION THREAD

**Discussion by nextleafzhao** [firefly-iii/firefly-iii#5849] [Firefly III]
When I followed the tutorial(https://docs.firefly-iii.org/firefly-iii/installation/self_hosted/) and got to the last command of "Initialize the database"(`php artisan passport:install`), I got the following message:
```
...
Now executing firefly:instructions update
+------------------------------------------------------------------------------+
|                                                                              |
| Thank you for updating to Firefly III, v5.6.16!                              |
|                                                                              |
|                                                                              |
+------------------------------------------------------------------------------+
Now executing firefly-iii:verify-security-alerts
huang@nluserver:/var/www/html/firefly-iii$ sudo php artisan passport:install
Encryption keys already exist. Use the --force option to overwrite them.
Personal access client created successfully.
Client ID: 1
Client secret: Su4dr7mAblqx7egKVkTZm28W5lrl2TiIhfU3bfjC
Password grant client created successfully.
Client ID: 2
Client secret: 3RXZvw4BpS8IL4IAOj5IrDA7u33Pwk5bkRu7ql0j
huang@nluserver:/var/www/html/firefly-iii$ sudo php artisan passport:install --force
Encryption keys generated successfully.
Personal access client created successfully.
Client ID: 3
Client secret: L5H0I6DIAjrUWsFOafOQAAnmaZO0DaWcMoSHY4rQ
Password grant client created successfully.
Client ID: 4
Client secret: 7wG8hE8x6RcUODP6lGoMzoW78uZnesUPCpAlw7eV
```
Then I used the `--force` parameter to execute it.And visit 
`http://localhost/firefly-iii/`
A large `403 Forbidden` appears !

I thought it was a file permissions problem, but after repeating the two commands it was still 403.
```
sudo chown -R www-data:www-data firefly-iii
sudo chmod -R 775 firefly-iii/storage
```

---

What should I do?

my partial `.env` file content:
```
DB_CONNECTION=mysql
DB_HOST=127.0.0.1
DB_PORT=3306
DB_DATABASE=firefly
DB_USERNAME=fireflyuser
DB_PASSWORD=xW9CoF8A
```

fireflyuser is authorized:
```sql
grant all on firefly.* to "fireflyuser"@"localhost" with grant option;
```

MySQL version: 8.0.
PHP version: 8.1.3

The info.php page looks normal.

**Comment by JC5** [comment]
> Then I used the --force parameter to execute it.

Not necessary, but it doesn't matter much.

> http://localhost/firefly-iii/

Browse to /firefly-iii/public, that could help!

**Reply by nextleafzhao** [reply]
Uh oh. Browse to /firefly-iii/public, it's still 403 Forbidden

**Reply by JC5** [reply]
Sorry, in that case I'm out of ideas. You could try hitting /firefly-iii/public/index.php, that would at least indicate Firefly III is up and running (or not).

**Reply by nextleafzhao** [reply]
I tried a different PHP version, now version 8.0, but....

Visit

        http://localhost/firefly-iii/

I got this:

Danger! This directory should not be open to the public!

/public/ should be the document root of your web server.

Leaving your web server configured like this is a huge security risk.

Please read more on the Github help pages.

After a few seconds it becomes 404

**Reply by JC5** [reply]
Make sure the root is /public/
